system-configuration/src/network_configuration.rs

```diff
@@ -0,0 +1,452 @@
+// Copyright 2017 Amagicom AB.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Bindings to [`SCNetworkConfiguration`].
+//!
+//! See the examples directory for examples how to use this module.
+//!
+//! [`SCNetworkConfiguration`]: https://developer.apple.com/documentation/systemconfiguration/scnetworkconfiguration
+
+use core_foundation::array::CFArray;
+use core_foundation::base::{CFType, TCFType};
+use core_foundation::base::kCFAllocatorDefault;
+use core_foundation::dictionary::CFDictionary;
+use core_foundation::string::{CFString, CFStringRef};
+
+use dynamic_store::SCDynamicStoreBuilder;
+pub use system_configuration_sys::network_configuration::*;
+use system_configuration_sys::preferences::SCPreferencesCreate;
+
+use std::{fmt, ptr};
+use std::net::IpAddr;
+
+/// MTU
+#[derive(Debug)]
+pub struct SCNetworkInterfaceMTU {
+    /// the current MTU setting for the interface.
+    pub current: u32,
+    /// the minimum MTU setting for the interface. If negative, the minimum setting could not
+    /// be determined.
+    pub min: Option<u32>,
+    /// the maximum MTU setting for the interface. If negative, the maximum setting could not
+    /// be determined.
+    pub max: Option<u32>,
+}
+
+/// DNS
+#[derive(Debug)]
+pub struct SCNetworkServiceDNS {
+    state_domain_name: Option<String>,
+    setup_domain_name: Option<String>,
+    state_server_addresses: Option<Vec<IpAddr>>,
+    setup_server_addresses: Option<Vec<IpAddr>>,
+}
+
+impl SCNetworkServiceDNS {
+    /// DNS Constructor
+    pub fn new(
+        domain_name: (Option<String>, Option<String>),
+        server_addresses: (Option<Vec<IpAddr>>, Option<Vec<IpAddr>>),
+    ) -> SCNetworkServiceDNS {
+        SCNetworkServiceDNS {
+            state_domain_name: domain_name.0,
+            setup_domain_name: domain_name.1,
+            state_server_addresses: server_addresses.0,
+            setup_server_addresses: server_addresses.1,
+        }
+    }
+    
+    /// Returns DomainName (state and setup)
+    pub fn domain_name(&self) -> (Option<String>, Option<String>) {
+        (
+            self.state_domain_name.clone(),
+            self.setup_domain_name.clone(),
+        )
+    }
+
+    /// Returns ServerAddresses (state and setup)
+    pub fn server_addresses(&self) -> (Option<Vec<IpAddr>>, Option<Vec<IpAddr>>) {
+        (
+            self.state_server_addresses.clone(),
+            self.setup_server_addresses.clone(),
+        )
+    }
+}
+
+/// Global network object
+pub struct SCNetworkGlobal;
```

Comment:
Okay

Reply:
> The query method could just be a standalone function in this module since it would be required in more than one struct impl.

that's two method `query` have diffence type.